src/main/resources/mapper/RouteMapper.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN"
+        "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
+<mapper namespace="org.example.dao.RouteDAO">
+    <insert id="insertRoute" keyColumn="id" keyProperty="id" useGeneratedKeys="true">
+        INSERT INTO routes (id, start_point, end_point, distance) VALUES (#{id}, #{startPoint}, #{endPoint}, #{distance})
+    </insert>
+    <update id="updateRoute">
+        UPDATE routes SET distance WHERE id = #{id}
+    </update>
+    <resultMap id="routesResult" type="org.example.model.Route">
+        <result property="id" column="id"/>
+        <result property="startPoint" column="start_point"/>
+        <result property="endPoint" column="end_point"/>
```

Comment:
since these two are classes within the Route class and not primitives, I think they should be done using something like 
association property="startPoint" resultMap="org.example.dao.PointDAO.PointResultMap"/

Reply:
Let's discuss that during the meeting